File: Supermercado/supermercado1.cs
//Supermercado con regex C Sharp
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

namespace Supermercado
{
    class Supermercado
    {
        static void Main(string[] args)
        {
            //def
            string cad, regex;
            int conta = 0;

            while (true)
            {
                cad = Console.ReadLine()!;

                if (string.IsNullOrEmpty(cad))
                {
                    Environment.Exit(0);
                }
                else
                {
                    //cadenas.Add(cad);
                    regex = Regex.Replace(cad, @"\s+", " ").TrimStart();
                    string[] nums = regex.Split(' ');

                    conta = 0;
                    for (int j = 0; j < nums.Length; j++)
                    {
                        conta = conta + int.Parse(nums[j]);
                    }
                    Console.WriteLine(conta);
                }
            }
        }
    }
}

File: Supermercado/Supermercado1.py
#supermercado python con regex
import sys
import re

conta = int(0)
while True:
    try:
      cad = input()
    except EOFError:
      break
    
    if not cad:
        # la cadena vacia...
        sys.exit()
    else:
        cad_strip = cad.strip()
        cad_split = re.split("\s+", cad_strip)
        conta = 0;
        for i in range(0, len(cad_split), 1):
            conta = conta + int(cad_split[i])
        print(conta)
#end program